### PR TITLE
Add basic stats for strongly consistent operations

### PR DESCRIPTION
**Pull request by jtuple**
Consistent gets/puts now correctly increment `vnode_gets`/`vnode_puts` the same as normal gets/puts. Prior to this commit, gets incremented the relevant stat but puts did not.

Consistent operations now maintain per-operation stats that are similar to the node/FSM stats for normal gets/puts. This includes operation counts, latency percentiles, and object size percentiles.

Future work is needed to support per-bucket stats, detailed `riak_ensemble` stats that track FSM stages / interesting events (eg. object rewrites after leader change), etc. But, these basic stats should be good enough for 2.0.

Addresses #876

### CODE REVIEW

File: src/riak_client.erl [hunk 1]
```diff
@@ -97,16 +97,36 @@ consistent_get(Bucket, Key, Options, {?MODULE, [Node, _ClientId]}) ->
...
+                             {ok, Obj}
+                     end
+             end,
+    maybe_update_consistent_stat(Node, consistent_get, Bucket, StartTS, Result),
+    Result.
+
+maybe_update_consistent_stat(Node, Stat, Bucket, StartTS, Result) ->
+    case node() of
+        Node ->
+            Duration = timer:now_diff(os:timestamp(), StartTS),
```
Comment (russelldb) [inline src/riak_client.erl:118]: in other places this has led to negative numbers being passed to folsom, which silently writes them and later dies on calculation: https://github.com/basho/riak_kv/issues/452. Not sure if we want to address it here, or hope that folsom/riak_kv_stat is patched up soon.

File: src/riak_kv_stat.erl [hunk 2]
```diff
@@ -274,10 +274,26 @@ do_update({fsm_destroy, Type}) ->
...
+    ok = notify_existing({?APP, consistent, gets, time}, Microsecs, histogram),
+    ok = maybe_notify_existing({?APP, consistent, gets, objsize}, ObjSize, histogram);
+do_update({consistent_put, _Bucket, Microsecs, ObjSize}) ->
+    ok = notify_existing({?APP, consistent, puts}, 1, spiral),
+    ok = notify_existing({?APP, consistent, puts, time}, Microsecs, histogram),
+    ok = maybe_notify_existing({?APP, consistent, puts, objsize}, ObjSize, histogram).
 
 %% private
 
+notify_existing(Name, Event, Type) ->
```
Comment (russelldb) [inline src/riak_kv_stat.erl:289]: didn't feel like refactoring all the other calls to `folsom_metrics:notify_existing_metric/3` eh? Neither would I, tbh.

File: src/riak_kv_stat.erl [hunk 3]
```diff
@@ -477,7 +493,13 @@ stats() ->
...
      {[object, map, merge, time], histogram},
-     {late_put_fsm_coordinator_ack, counter}
+     {late_put_fsm_coordinator_ack, counter},
+     {[consistent, gets], spiral},
+     {[consistent, gets, time], histogram},
+     {[consistent, gets, objsize], histogram},
+     {[consistent, puts], spiral},
+     {[consistent, puts, time], histogram},
+     {[consistent, puts, objsize], histogram}
     ].
```
Comment (russelldb) [inline src/riak_kv_stat.erl:504]: missing the corresponding entries in `stats_from_update_arg`, But I''ll do that 'cos it's my fault we have such horror here.

File: src/riak_kv_stat_bc.erl [hunk 4]
```diff
@@ -439,6 +440,35 @@ legacy_put_fsm_stats() ->
             []
     end.
 
+consistent_stats() ->
```
Comment (russelldb) [inline src/riak_kv_stat_bc.erl:443]: I'm sorry :(

**Comment by russelldb** [comment]
Made some comments, added a commit, none of the commente really need acting on, just in case you want to talk about it or something.

I'm +1 on the changes, but looking into that dialyzer error

**Comment by engelsanchez** [comment]
@russelldb I've merged and taken care of the dialyzer errors. Mind giving the 404cb0e commit a plus one now?

**Comment by russelldb** [comment]
+1 404cb0e

**Comment by jonmeredith** [comment]
@engelsanchez are we good to merge?

**Comment by engelsanchez** [comment]
We were at some point, but Joe didn't merge. Joe will just have to get a clean build again by resolving the merge conflicts. Feature-wise it seems yes.  I'm not the reviewer or anything. I just tried to resolve the previous conflicts so Joe could just push a button and be done with this.

**Comment by jtuple** [comment]
@borshop: retry

**Comment by reiddraper** [comment]
@jtuple retrying won't do anything, this branch doesn't merge-cleanly with develop.

**Comment by engelsanchez** [comment]
@jtuple unfortunately now there is a merge conflict with the dialyzer warnings changes  :`(.

**Comment by engelsanchez** [comment]
Since @jtuple seems OK with merging this PR as is, I'll fix the conflicts and initiate the merge when it's ready and passing again.

**Comment by jtuple** [comment]
@engelsanchez Thanks! Get 'er merged!

**Comment by engelsanchez** [comment]
+1 fc52f20

**Comment by engelsanchez** [comment]
@borshop merge

**Pull request merged**